composer/workflows/airflow_db_cleanup.py

```diff
@@ -319,37 +315,27 @@ def build_query(
     airflow_db_model,
     age_check_column,
     max_date,
-    keep_last,
-    keep_last_filters=None,
-    keep_last_group_by=None,
+    dag_id=None
 ):
     query = session.query(airflow_db_model)
 
     logging.info("INITIAL QUERY : " + str(query))
 
-    if not keep_last:
-        query = query.filter(
-            age_check_column <= max_date,
-        )
+    if airflow_db_model == DagRun:
+        newest_dagrun = session.query(airflow_db_model) \
+            .filter(airflow_db_model.dag_id == dag_id) \
+            .order_by(airflow_db_model.execution_date).first()
```

Comment:
Should it be actually .last()?

Reply:
Indeed I had 'desc' function here, but it got lost at some point when refactoring. Thank you!